stateth/stateth.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package stateth
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+	gapi "github.com/teemupo/go-grafana-api"
+	"gopkg.in/urfave/cli.v1"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	influxDBAdminUser = "test"  // admin username for InfluxDB.
+	influxDBAdminPass = "test"  // admin password for InfluxDB.
+	grafanaUser       = "admin" // default Grafana username - should not be changed here without first updating the docker image.
+	grafanaPass       = "admin" // default Grafana password - should not be changed here without first udpating the docker image.
+)
+
+// Stateth contains the stateth internals.
+type Stateth struct {
+	ctx     *cli.Context
+	config  *Config
+	gclient *gapi.Client // Grafana client.
+}
+
+// New creates a new stateth instance with the given configuration.
+func New(ctx *cli.Context, config *Config) (*Stateth, error) {
```

Comment:
Nitpick, but: I don't think `New` is part of any interface, so I see no reason for it to return an `error`. I can see that you are using it to `Register` a `Service`, but the `nil` error should be returned at that point.